tasks/main.yml

```
@@ -107,39 +87,80 @@
     manager: auto
   no_log: true
 
+- name: Verify if the mssql_version variable is provided correctly
+  assert:
+    that: >-
+      mssql_version | int == 2017
+      or mssql_version | int == 2019
+      or mssql_version | int == 2022
+    fail_msg: >-
+      You must set the mssql_version variable to one of
+      {{ __mssql_supported_versions | join(", ") }}
+
+- name: Verify that version 2022 is not used with RHEL 7
+  fail:
+    msg: >-
+      SQL Server 2022 does not support EL 7 hosts
+  when:
+    - mssql_version | int == 2022
+    - ansible_distribution in ['CentOS', 'RedHat']
+    - ansible_distribution_major_version is version('8', '<')
+
 - name: >-
     Verify if mssql_version is not smaller then the existing SQL Server version
   fail:
     msg:
-      - "You set mssql_version to 2017, but your SQL Server is version 2019."
-      - "The role does not support downgrading SQL Server."
-  when:
-    - mssql_version | int == 2017
-    - ansible_facts.packages["mssql-server"][0]["version"] is defined
-    - ansible_facts.packages["mssql-server"][0]["version"] is search("^15.*")
+      - >-
+        "You set mssql_version to {{ mssql_version }}, but your SQL Server is
+        of higher version"
+      - "The role does not support downgrading SQL Server"
+  when: >-
+    (mssql_version | int in [2017, 2019]
+    and
+    ansible_facts.packages["mssql-server"][0]["version"]
+    | d('0') is version('16', '>'))
+    or
+    (mssql_version | int == 2017
+    and
+    ansible_facts.packages["mssql-server"][0]["version"]
+    | d('0') is version('15', '>'))
 
 - name: Deploy the GPG key for Microsoft repositories
   rpm_key:
     key: "{{ mssql_rpm_key }}"
     state: present
 
-# This works only on systems that use yum or dnf
-- name: Update all packages from SQL Server 2017 repo
-  package:
-    name: "*"
-    disablerepo: "*"
-    enablerepo: packages-microsoft-com-mssql-server-2017
-    state: latest  # noqa 403
-  when:
-    - mssql_upgrade | bool
-    - ansible_facts.packages["mssql-server"][0]["version"] is defined
-    - ansible_facts.packages["mssql-server"][0]["version"] is search("^14.*")
-
-- name: Remove the Microsoft SQL Server 2017 repository to upgrade to 2019
-  yum_repository:
-    name: packages-microsoft-com-mssql-server-2017
-    state: absent
+- name: Prepare for the upgrade
   when: mssql_upgrade | bool
+  block:
+    # This works only on systems that use yum or dnf
+    - name: Update all packages from SQL Server 2017 repo
+      package:
+        name: "*"
+        disablerepo: "*"
+        enablerepo: packages-microsoft-com-mssql-server-2017
+        state: latest  # noqa 403
```

```suggestion
        state: latest  # noqa package-latest
```
use the string name instead of the number - https://ansible-lint.readthedocs.io/rules/package-latest/
I realize this was probably a copy/paste from somewhere else